superset/datasets/dao.py

```diff
@@ -190,28 +185,37 @@ def update_columns(
         - If there are extra columns on the metadata db that are not defined on the List
         then we delete.
         """
+
+        column_by_id = {column.id: column for column in model.columns}
         new_columns = []
+
         for column in property_columns:
             column_id = column.get("id")
 
             if column_id:
-                column_obj = db.session.query(TableColumn).get(column_id)
```

Comment:
Leverage the columns defined by the model rather than re-querying for each column.